File: strategy/marvin/agent/manual.lua
local Manual = (require "../base/class").new("Agent.Manual", require "agent/base/agent")

local World = require "../base/world"
local UserInput = require "../base/userinput"

local Default = require "agent/defender/default"

Manual._behaviors = {
	Default
}

function Manual.takeRobot(robots)
	for _, robot in pairs(robots) do
		-- take robots which get command from an input device
		local input = UserInput.getControlInput(robot)
		if input then
			return robot
		end
	end
end

function Manual:keepRobot()
	return UserInput.getControlInput(robot)
end

function Manual:rateRobot()
	return 0
end

return Manual

File: strategy/typescript/glados/agent/manual.ts
import { FriendlyRobot } from "base/robot";

import { Agent } from "glados/agent/base/agent";
import { Behavior } from "glados/agent/base/behavior";
import { Default } from "glados/agent/manual/default";


export class Manual extends Agent {

	public getBehaviors(): any[] {
		return [Default];
	}

	public static takeRobot(robots: FriendlyRobot[]): FriendlyRobot | undefined {
		for (let robot of robots) {
			// take robots which get command from an input device
			if (robot.userControl) {
				return robot;
			}
		}
		return undefined;
	}

	public keepRobot(): boolean {
		return this._robot.userControl != undefined;
	}

	public rateRobot(): number {
		return 0;
	}
}
